File: ws/gudino/u1/ws06Json/model/chicken_coop.py
from .chicken import Chicken

class ChickenCoop:
    def __init__(self, coop_id):
        self.id = coop_id
        self.chickens = []

    def add(self, chicken):
        self.chickens.append(chicken)

    def remove(self, chicken_id):
        self.chickens = [c for c in self.chickens if c.id != chicken_id]

    def count_chickens(self):
        return len(self.chickens)

    def show_all_chickens(self):
        if not self.chickens:
            print("No chickens in this coop.")
        else:
            for chicken in self.chickens:
                print(chicken)

    def to_dict(self):
        """Convierte a diccionario (para JSON)."""
        return {
            "coop_id": self.id,
            "chickens": [chicken.to_dict() for chicken in self.chickens]
        }

    def __str__(self):
        return f"ChickenCoop(id={self.id}, chickens={len(self.chickens)})"

File: hw/gudino/u1/hw11ChikenCoops/Chicken_Coop_JavaScript/model/chickenCoop.js
class ChickenCoop {
    constructor(id) {
        this.id = id;
        this.chickens = [];
    }

    add(chicken) {
        this.chickens.push(chicken);
    }

    remove(chickenId) {
        this.chickens = this.chickens.filter(c => c.id !== chickenId);
    }

    showAllChickens() {
        if (this.chickens.length === 0) {
            console.log("No chickens in this coop.");
        } else {
            this.chickens.forEach(c => console.log(c.toString()));
        }
    }

    toArray() {
        return this.chickens.map(c => [this.id, ...c.toArray()]);
    }
}

module.exports = ChickenCoop;
